core/elements/ons-switch.es6

```diff
@@ -68,16 +90,34 @@ limitations under the License.
     _compile() {
       this.classList.add('switch');
       this.innerHTML = `
-        <input type="checkbox" checked class="switch__input">
+        <input type="checkbox" class="switch__input">
         <div class="switch__toggle"></div>
       `;
       this._getCheckbox().setAttribute('name', generateId());
     }
 
     detachedCallback() {
+      this._getCheckbox().removeEventListener('change', this.onChangeListener);
     }
 
-    atachedCallback() {
+    attachedCallback() {
+      this._getCheckbox().addEventListener('change', this.onChangeListener);
+    }
+
+    onChangeListener() {
+      if (this.checked !== true) {
+        this.parentNode.removeAttribute('checked');
+      }
+      else {
+        this.parentNode.setAttribute('checked', '');
+      }
+
+      if (this.disabled !== true) {
```

Comment:
This statement can be removed. The value of `checkbox.disabled` will not change because the value of the checkbox changes.